monai/utils/sliding_window_inference.py

```diff
@@ -0,0 +1,165 @@
+        sw_batch_size (int): the batch size to run window slices.
+        predictor: a portion from numpy.lib.arraypad.pad is copied below.
+        device: on which device to execute model inference, cpu or gpu.
+
+    Note:
+        must be channel first, support both 2D and 3D.
+        input data must have batch dim.
+        execute on 1 image/per inference, run a batch of window slices of 1 input image.
+    """
+    num_spatial_dims = len(inputs.shape) - 2
+    assert len(roi_size) == num_spatial_dims, 'roi_size {} does not match input dims.'.format(roi_size)
+
+    # determine image spatial size and batch size
+    # Note: all input images must have the same image size and batch size
+    image_size = list(inputs.shape[2:])
+    batch_size = inputs.shape[0]
+
+    # TODO: Enable batch sizes > 1 in future.
+    assert batch_size == 1, "input batch size must be 1"
+
+    # in case that image size is smaller than roi size
+    is_oversized = False
+    original_image_size = [image_size[i] for i in range(num_spatial_dims)]
+
+    for i in range(num_spatial_dims):
+        if int(roi_size[i]) > image_size[i]:
+            is_oversized = True
+            break
+
+    if is_oversized:
+        for i in range(num_spatial_dims):
+            image_size[i] = max(image_size[i], roi_size[i])
+
+        padder = ImageEndPadder(roi_size, 'constant')
+        inputs = padder(inputs)
+
+    scan_interval = _get_scan_interval(image_size, roi_size, num_spatial_dims)
+    scan_num = [int(math.ceil(float(image_size[i]) / scan_interval[i])) for i in range(num_spatial_dims)]
+
+    # Store all slices in list.
+    slices = []
+    if num_spatial_dims == 3:
+        for i in range(scan_num[0]):
+            start_i = i * scan_interval[0]
+            start_i -= max(start_i + roi_size[0] - image_size[0], 0)
+            slice_i = slice(start_i, start_i + roi_size[0])
+
+            for j in range(scan_num[1]):
+                start_j = j * scan_interval[1]
+                start_j -= max(start_j + roi_size[1] - image_size[1], 0)
+                slice_j = slice(start_j, start_j + roi_size[1])
+
+                for k in range(0, scan_num[2]):
+                    start_k = k * scan_interval[2]
+                    start_k -= max(start_k + roi_size[2] - image_size[2], 0)
+                    slice_k = slice(start_k, start_k + roi_size[2])
+                    slices.append((slice_i, slice_j, slice_k))
+    else:
+        for i in range(scan_num[0]):
+            start_i = i * scan_interval[0]
+            start_i -= max(start_i + roi_size[0] - image_size[0], 0)
+            slice_i = slice(start_i, start_i + roi_size[0])
+
+            for j in range(scan_num[1]):
+                start_j = j * scan_interval[1]
+                start_j -= max(start_j + roi_size[1] - image_size[1], 0)
+                slice_j = slice(start_j, start_j + roi_size[1])
+                slices.append((slice_i, slice_j))
+
+    buffered_requests = []
```

Comment:
@Nic-Ma A comment here to state intention of this code block would be useful. Maybe "Stack slices for batch inference". Also "buffered_requests" should be changed to something more readable like batched slices?

Reply:
@wyli already made a lot of changes in this PR: #56 .
You can suggest Wenqi to change the variable name directly if necessary.
Thanks.

Reply:
thanks @madil90  I'm chaning it in #56